src/main/java/org/testng/ITestNGMethod.java

```diff
@@ -154,7 +154,17 @@
 
   ITestNGMethod clone();
 
-  IRetryAnalyzer getRetryAnalyzer();
+  /**
+   * @deprecated - This method stands deprecated as of TestNG 7.0.0
+   */
+  @Deprecated

```

Comment:
I didn't understand both these steps. Can you please elaborate ?

> With your solution, we miss an important part of interface : force method implementation

True that. I can get rid of the `default` implementation if that makes more sense, and thus force method implementation.

Reply:
The objective of such modification is to change api without breaking things: 
we want to use the new method, but we can't add it in interfaces (default method is the clue) and we can't remove the old method (because we use it in the default method).
So we just have to deprecate the old one in the first step.

Some versions later, we can assume the breaking change is known and we can remove the deprecated method and replace the default impl by the signature.